pkg/agent/util/net_windows.go

```diff
@@ -619,6 +634,22 @@ func SetAdapterDNSServers(adapterName, dnsServers string) error {
 	return nil
 }
 
+// GetDHCPByInterfaceIndex returns the DHCP status on the specified interface.
+func GetDHCPByInterfaceIndex(ifIndex int) (bool, error) {
```

Comment:
You can use syscall function `GetIPInterfaceEntry` by checking property "ManagedAddressConfigurationSupported" to reduce the additional cost introduced by powershell.